gcp/ignore_error_predicate.go

```diff
@@ -20,15 +21,22 @@ func isIgnorableError(notFoundErrors []string) plugin.ErrorPredicate {
 	}
 }
 
-// shouldIgnoreErrorPluginDefault:: Plugin level default function to ignore a set errors for hydrate functions based on "ignore_error_codes" config argument
+// shouldIgnoreErrorPluginDefault:: Plugin level default function to ignore a set errors for hydrate functions based on "ignore_error_codes" and "ignore_error_messages" config argument
 func shouldIgnoreErrorPluginDefault() plugin.ErrorPredicateWithContext {
 	return func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, err error) bool {
-		if !hasIgnoredErrorCodes(d.Connection) {
-			return false
-		}
-
 		gcpConfig := GetConfig(d.Connection)
+
 		if gerr, ok := err.(*googleapi.Error); ok {
+
+			// Add to support regex match as per error message
+			for _, pattern := range gcpConfig.IgnoreErrorMessages {
+				re := regexp.MustCompile(pattern)
+				result := re.MatchString(gerr.Message)
+				if result {
+					return true
+				}
+			}
+
```

Comment:
Turns out not all Google Cloud SDK APIs use the `googleapi.Error` type, e.g. Redis APIs, so I've altered this part of the code to compare the error message to all errors:

```suggestion

		logger := plugin.Logger(ctx)
		logger.Trace("ignore_error_predicate.shouldIgnoreErrorPluginDefault", "err.Error()", err.Error())

		// Add to support regex match as per error message
		for _, pattern := range gcpConfig.IgnoreErrorMessages {
			logger.Trace("ignore_error_predicate.shouldIgnoreErrorPluginDefault", "message_pattern", pattern)
			// TODO: maybe compile these at the connection level for performance reasons
			re := regexp.MustCompile(pattern)
			result := re.MatchString(err.Error())
			if result {
				logger.Debug("ignore_error_predicate.shouldIgnoreErrorPluginDefault", "ignore_error_message", err.Error())
				return true
			}
		}

```

Without this change, `API has not been used` cannot be ignore for Redis API calls.

Note that this requires enabling multiline regex matching with `(?s)`, e.g.:
```
                ignore_error_messages = [
                    "(?s)^.*API has not been used.*$",
                    "(?s)^.*Unsupported region.*$",
                ]
```

Also added some logs to the error code checks right below.

Pushed the changes here: https://github.com/pdecat/steampipe-plugin-gcp/commit/d8b31a212dc54f93f40805f2994c744e314b7e3e

Can submit an PR targeting this PR's branch if you want.

Reply:
@pdecat Yes please, we'd greatly appreciate a PR!